src/search/SearchMessagesScreen.js

```diff
@@ -71,37 +73,39 @@ class SearchMessagesScreen extends PureComponent<Props, State> {
   // invalidate outstanding requests on change will require more work.
 
   handleQueryChange = async (query: string) => {
-    const id = ++this.lastIdSent;
-
-    if (query === '') {
-      // The empty query can be resolved without a network call.
-      this.lastIdReceived = id;
-      this.lastIdSuccess = id;
-      this.setState({ messages: null, isFetching: false });
-      return;
-    }
-
-    this.setState({ isFetching: true });
-    try {
-      const messages = await this.fetchSearchMessages(query);
-
-      // Update `state.messages` if this is our new latest result.
-      if (id > this.lastIdSuccess) {
+    this.setState({ query }, async () => {
```

Comment:
I did split the commit in new revision, but along with the edit you suggested at https://github.com/zulip/zulip-mobile/pull/4900#discussion_r670934688, it makes the commit that just adds query to component's state raise an eslint error for no unused state. I am supressing this error since using it would essentially be the part two of the split: "Making this component's child, SearchMessagesCard, take query as a prop."